test/phoenix_live_view/html_formatter_test.exs

```diff
@@ -448,7 +448,7 @@ if Version.match?(System.version(), ">= 1.13.0") do
 
         """,
         """
-        <button class="btn-primary" autofocus disabled>Submit</button>
+        <button class="btn-primary" autofocus disabled> Submit </button>
```

Comment:
Particularly, I don't link this since here we don't expect spaces. Also, does it cover the case in the issue?

```elixir
      # does not remove the whitepace from `<b>Foo: </b<>`
      # because there isn't whitespace in the upcoming text.
      assert_formatter_doesnt_change("""
      <p>
        <b>Foo: </b><%= some_var %>
      </p>
      """)
```

Another potential solution is something around where:

https://github.com/phoenixframework/phoenix_live_view/blob/6f4d1b1b523e2eda1f5858ce266931a65cdfa6b5/lib/phoenix_live_view/html_formatter.ex#L543

We might be able to add more conditions such as checking if the state of the last text in the tag_block so we can either preserve or not accordingly. I gotta play a bit more myself though to make sure it will cover it all 🤔

Reply:
> does it cover the case in the issue?

Yes 👍🏻 

> I gotta play a bit more myself though to make sure it will cover it all

Sure! Hopefully that case can be handled by the formatter.

Reply:
Unfortunately my approach didn't work 😢 

What if we do it **only** for inline tags that are common to have texts next to each other? I mean something like this:

```elixir
          context =
            # maybe more tags in here
            if name not ~w(button) and inline_text?(block, meta.mode) do
              %{context | mode: :inline}
            else
              context
            end
```

For instance, the spaces within buttons doesn't not interfere in the render. So I guess we could totally skip it for this logic - maybe others tags listed here too:

https://github.com/phoenixframework/phoenix_live_view/blob/6f4d1b1b523e2eda1f5858ce266931a65cdfa6b5/lib/phoenix_live_view/html_formatter.ex#L193


cc @josevalim

Reply:
Unfortunately my approach didn't work 😢 

What if we do it **only** for inline tags that are common to have texts next to each other? I mean something like this:

```elixir
          context =
            # maybe more tags in here
            if name not ~w(button) and inline_text?(block, meta.mode) do
              %{context | mode: :inline}
            else
              context
            end
```

For instance, the spaces within buttons doesn't not interfere in the render. So I guess we could totally skip it for this logic - maybe others tags listed here too:

https://github.com/phoenixframework/phoenix_live_view/blob/6f4d1b1b523e2eda1f5858ce266931a65cdfa6b5/lib/phoenix_live_view/html_formatter.ex#L193


cc @josevalim

Reply:
> What if we do it only for inline tags that are common to have texts next to each other?

Yeah I agree, we can fine tune that logic.